ui/lib/core/addon/templates/components/icon.hbs

```diff
@@ -1,3 +1,12 @@
-<span class="hs-icon {{sizeClass}} {{class}}" ...attributes>
-  {{svg-jar @glyph}}
-</span>
\ No newline at end of file
+{{#if this.isFlightIcon}}
+  <FlightIcon
+    @name={{@name}}
+    @size={{this.flightIconSize}}
+    class={{this.iconClass}}
```

Comment:
Is `iconClass` still relevant to flight icons, since we pass in the size attribute?

Reply:
It is in order to keep the styling consistent between icon sets. Currently flight icons only support 16 and 24 for the `size` param but there are 6 different sizes for the _hs-icon_ class. I created `flight-icon.scss` to tweak the _hs-icon_ size classes to match as closely as possible which is then applied to the component.

I wrestled with this a bit but wanted to support the `size` parameter for _FlightIcon_ while still preserving the _hs-icon_ sizing that existed. If it's confusing we could make `size` private for now and rely on the _hs-icon_ size classes until _FlightIcon_ size becomes more flexible.

Reply:
I wonder if we should instead conform our sizes to the new standard 🤔 may want to wait for Ivana to weigh in on this one

Reply:
@ivana-mcconnell what are your thoughts on this? You can see how the sizes are mapped in [this commit](https://github.com/hashicorp/vault/pull/12976/commits/a1fb58d00ca321b5c51a77257a90df9514910845#diff-74a077143fbd0008b419aa84c0809ca816c9c3496eddbcaddc7ee898a56e32cf) and here's a breakdown of the number of times the _Icon_ component is used in the app across the various sizes:

s - 7
m - 118
l - 37
xlm - 3
xl - 11
xxl - 1

Reply:
I'm on board with conforming our sizes to the new standard, but I think that it's worth a pairing session to look at the instances where icons are _not_ 16px or 24px and making adjustments for them, just so we don't end up with something incredibly disproportionate. It looks like that's a total of 22 instances (s, xlm, xl, xxl). Might be a little tedious, but probably worthwhile.

Reply:
@hashishaw after pairing with Ivana and confirming the new size standards I updated the component and simplified things in [this commit](https://github.com/hashicorp/vault/pull/12976/commits/5aa1cd3f294d748587cdda076e088ea1defbf100).